source/core_http_client.c

```diff
@@ -574,10 +577,23 @@ static int8_t caseInsensitiveStringCmp( const char * str1,
                                         size_t n )
 {
     size_t i = 0U;
+    /* Inclusion of inbetween variables for coverity rule 13.2 compliance */
+    int32_t firstChar;
+    int32_t secondChar;
 
     for( i = 0U; i < n; i++ )
     {
-        if( toupper( str1[ i ] ) != toupper( str2[ i ] ) )
+        /* MISRA Ref 21.13.1 [Essential type casting] */
+        /* More details at: https://github.com/FreeRTOS/coreHTTP/blob/main/MISRA.md#rule-2113 */
+        /* coverity[misra_c_2012_rule_21_13_violation] */
+        firstChar = ( int32_t ) toupper( ( ( int32_t ) str1[ i ] ) );
```

Comment:
Why is a deviation needed for this?

Reply:
The rule is being flagged since there isn't a cast to an unsigned char before passing the value str1[i] to toupper(). If I add the cast to do that then it starts to fail the CBMC proofs.

Reply:
That seems to be an issue with CBMC then. Are we sure we are not masking a real issue?

Also is the outer cast needed? I was under the impression that MISRA allowed use of int when dealing with standard lib function return values.

Reply:
Yeah the extra cast was left accidentally, I've since removed it.
I hadn't dug incredibly deeply into the cause of the CBMC failure if casting the char to an unsigned, I can take some time and look further into it if that's what you want.